src/Resolver/NameResolver/NameResolver.php

```diff
@@ -31,10 +32,10 @@ public function resolve($node): ?string
         if ($node instanceof Name || $node instanceof Identifier) {
             return $this->resolve((string)$node);
         }
-        if ($node instanceof Variable) {
+        if ($node instanceof Variable && !$node->name instanceof Expr) {
```

Comment:
why this condition here? if you pass it to resolve method, it will evaluated to null because no if will match

Reply:
`$$varName` wit otherwise resolve as `$varName` not as unknown

Reply:
isn't it case in all condition blocks?

Reply:
What do you mean by condition blocks? BUt theoretically it can happend enywhere variable-variable can occur. So even a method name

Reply:
I thought in each if in this class :) so I think solution should be add if ($node instanceof Expr) { return null; } at the beginging of resolve method or after 
```
    if ($node === null) {
            return null;
        }
        if (is_string($node)) {
            return $node !== '' ? $node : null;
        }
```

Reply:
That would make resolving Variable impossible because Variable is Expr.

Reply:
So this condition should be for every call to $this->resolve in this class

Reply:
Or maybe add something like `resolveNext` that would contain this safeguard and then call resolve. And in all branches we should call only resolveNext

Reply:
resolveNext could be like this:
```
private function resolveNext(Node $node)
{
if ($node instanceof Expr) {
   return null;
}
return $this->resolve($node);
```

right?

Reply:
Yes exactly